File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-com-5-4/GasProduct.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class GasProduct : ISerialized
    {
        public GasProduct(XmlNode xmlNode)
        {
            XmlNode typeNode = xmlNode.SelectSingleNode("type");
            
            if (typeNode != null)
            {
                if (typeNode.Attributes["href"] != null || typeNode.Attributes["id"] != null) 
                {
                    if (typeNode.Attributes["id"] != null) 
                    {
                        typeIDRef_ = typeNode.Attributes["id"].Value;
                        GasProductTypeEnum ob = new GasProductTypeEnum(typeNode);
                        IDManager.SetID(typeIDRef_, ob);
                    }
                    else if (typeNode.Attributes["href"] != null)
                    {
                        typeIDRef_ = typeNode.Attributes["href"].Value;
                    }
                    else
                    {
                        type_ = new GasProductTypeEnum(typeNode);
                    }
                }
                else
                {
                    type_ = new GasProductTypeEnum(typeNode);
                }
            }
            
        
            XmlNode calorificValueNode = xmlNode.SelectSingleNode("calorificValue");
            
            if (calorificValueNode != null)
            {
                if (calorificValueNode.Attributes["href"] != null || calorificValueNode.Attributes["id"] != null) 
                {
                    if (calorificValueNode.Attributes["id"] != null) 
                    {
                        calorificValueIDRef_ = calorificValueNode.Attributes["id"].Value;
                        NonNegativeDecimal ob = new NonNegativeDecimal(calorificValueNode);
                        IDManager.SetID(calorificValueIDRef_, ob);
                    }
                    else if (calorificValueNode.Attributes["href"] != null)
                    {
                        calorificValueIDRef_ = calorificValueNode.Attributes["href"].Value;
                    }
                    else
                    {
                        calorificValue_ = new NonNegativeDecimal(calorificValueNode);
                    }
                }
                else
                {
                    calorificValue_ = new NonNegativeDecimal(calorificValueNode);
                }
            }
            
        
            XmlNode qualityNode = xmlNode.SelectSingleNode("quality");
            
            if (qualityNode != null)
            {
                if (qualityNode.Attributes["href"] != null || qualityNode.Attributes["id"] != null) 
                {
                    if (qualityNode.Attributes["id"] != null) 
                    {
                        qualityIDRef_ = qualityNode.Attributes["id"].Value;
                        GasQuality ob = new GasQuality(qualityNode);
                        IDManager.SetID(qualityIDRef_, ob);
                    }
                    else if (qualityNode.Attributes["href"] != null)
                    {
                        qualityIDRef_ = qualityNode.Attributes["href"].Value;
                    }
                    else
                    {
                        quality_ = new GasQuality(qualityNode);
                    }
                }
                else
                {
                    quality_ = new GasQuality(qualityNode);
                }
            }
            
        
        }
        
    
        #region Type_
        private GasProductTypeEnum type_;
        public GasProductTypeEnum Type_
        {
            get
            {
                if (this.type_ != null)
                {
                    return this.type_; 
                }
                else if (this.typeIDRef_ != null)
                {
                    type_ = IDManager.getID(typeIDRef_) as GasProductTypeEnum;
                    return this.type_; 
                }
                else
                {
                      return this.type_; 
                }
            }
            set
            {
                if (this.type_ != value)
                {
                    this.type_ = value;
                }
            }
        }
        #endregion
        
        public string typeIDRef_ { get; set; }
        #region CalorificValue_
        private NonNegativeDecimal calorificValue_;
        public NonNegativeDecimal CalorificValue_
        {
            get
            {
                if (this.calorificValue_ != null)
                {
                    return this.calorificValue_; 
                }
                else if (this.calorificValueIDRef_ != null)
                {
                    calorificValue_ = IDManager.getID(calorificValueIDRef_) as NonNegativeDecimal;
                    return this.calorificValue_; 
                }
                else
                {
                      return this.calorificValue_; 
                }
            }
            set
            {
                if (this.calorificValue_ != value)
                {
                    this.calorificValue_ = value;
                }
            }
        }
        #endregion
        
        public string calorificValueIDRef_ { get; set; }
        #region Quality_
        private GasQuality quality_;
        public GasQuality Quality_
        {
            get
            {
                if (this.quality_ != null)
                {
                    return this.quality_; 
                }
                else if (this.qualityIDRef_ != null)
                {
                    quality_ = IDManager.getID(qualityIDRef_) as GasQuality;
                    return this.quality_; 
                }
                else
                {
                      return this.quality_; 
                }
            }
            set
            {
                if (this.quality_ != value)
                {
                    this.quality_ = value;
                }
            }
        }
        #endregion
        
        public string qualityIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-com-5-4/GasProduct.cpp
// GasProduct.cpp 
#include "GasProduct.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

GasProduct::GasProduct(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //typeNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* typeNode = xmlNode->FirstChildElement("type");

   if(typeNode){typeIsNull_ = false;}
   else{typeIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- typeNode , address : " << typeNode << std::endl;
   #endif
   if(typeNode)
   {
      if(typeNode->Attribute("href") || typeNode->Attribute("id"))
      {
          if(typeNode->Attribute("id"))
          {
             typeIDRef_ = typeNode->Attribute("id");
             type_ = boost::shared_ptr<GasProductTypeEnum>(new GasProductTypeEnum(typeNode));
             type_->setID(typeIDRef_);
             IDManager::instance().setID(typeIDRef_,type_);
          }
          else if(typeNode->Attribute("href")) { typeIDRef_ = typeNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { type_ = boost::shared_ptr<GasProductTypeEnum>(new GasProductTypeEnum(typeNode));}
   }

   //calorificValueNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* calorificValueNode = xmlNode->FirstChildElement("calorificValue");

   if(calorificValueNode){calorificValueIsNull_ = false;}
   else{calorificValueIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- calorificValueNode , address : " << calorificValueNode << std::endl;
   #endif
   if(calorificValueNode)
   {
      if(calorificValueNode->Attribute("href") || calorificValueNode->Attribute("id"))
      {
          if(calorificValueNode->Attribute("id"))
          {
             calorificValueIDRef_ = calorificValueNode->Attribute("id");
             calorificValue_ = boost::shared_ptr<NonNegativeDecimal>(new NonNegativeDecimal(calorificValueNode));
             calorificValue_->setID(calorificValueIDRef_);
             IDManager::instance().setID(calorificValueIDRef_,calorificValue_);
          }
          else if(calorificValueNode->Attribute("href")) { calorificValueIDRef_ = calorificValueNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { calorificValue_ = boost::shared_ptr<NonNegativeDecimal>(new NonNegativeDecimal(calorificValueNode));}
   }

   //qualityNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* qualityNode = xmlNode->FirstChildElement("quality");

   if(qualityNode){qualityIsNull_ = false;}
   else{qualityIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- qualityNode , address : " << qualityNode << std::endl;
   #endif
   if(qualityNode)
   {
      if(qualityNode->Attribute("href") || qualityNode->Attribute("id"))
      {
          if(qualityNode->Attribute("id"))
          {
             qualityIDRef_ = qualityNode->Attribute("id");
             quality_ = boost::shared_ptr<GasQuality>(new GasQuality(qualityNode));
             quality_->setID(qualityIDRef_);
             IDManager::instance().setID(qualityIDRef_,quality_);
          }
          else if(qualityNode->Attribute("href")) { qualityIDRef_ = qualityNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { quality_ = boost::shared_ptr<GasQuality>(new GasQuality(qualityNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<GasProductTypeEnum> GasProduct::getType()
{
   if(!this->typeIsNull_){
        if(typeIDRef_ != ""){
             return boost::shared_static_cast<GasProductTypeEnum>(IDManager::instance().getID(typeIDRef_));
        }else{
             return this->type_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<GasProductTypeEnum>();
   }
}
boost::shared_ptr<NonNegativeDecimal> GasProduct::getCalorificValue()
{
   if(!this->calorificValueIsNull_){
        if(calorificValueIDRef_ != ""){
             return boost::shared_static_cast<NonNegativeDecimal>(IDManager::instance().getID(calorificValueIDRef_));
        }else{
             return this->calorificValue_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<NonNegativeDecimal>();
   }
}
boost::shared_ptr<GasQuality> GasProduct::getQuality()
{
   if(!this->qualityIsNull_){
        if(qualityIDRef_ != ""){
             return boost::shared_static_cast<GasQuality>(IDManager::instance().getID(qualityIDRef_));
        }else{
             return this->quality_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<GasQuality>();
   }
}
}

